src/CI/UrlChecker.php

```diff
@@ -11,21 +11,20 @@
 
 namespace SymfonyDocsBuilder\CI;
 
-use GuzzleHttp\Client;
-use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\HttpClient\HttpClient;
 
 final class UrlChecker
 {
     private $invalidUrls = [];
 
-    public function checkUrl(string $url)
+    public function checkUrl(string $url): void
     {
-        $httpClient = new Client(['timeout' => 10]);
+        $httpClient = HttpClient::create(['timeout' => 10]);
 
         try {
-            $response = $httpClient->get($url, ['http_errors' => false]);
+            $response = $httpClient->request('GET', $url);
             $statusCode = $response->getStatusCode();
-        } catch (GuzzleException $exception) {
+        } catch (\RuntimeException $e) {
```

Comment:
shouldn't we narrow catch to `\Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface`?

Reply:
I've changed it to `HttpExceptionInterface` ... if we catch only Transport exception we'd miss Server exception, Redirection exception, etc.

Reply:
indeed, i meant this one :+1: